src/components/ContentSidebar/ActivitySidebar.js

```diff
@@ -8,15 +8,23 @@ import * as React from 'react';
 import debounce from 'lodash/debounce';
 import noop from 'lodash/noop';
 import { FormattedMessage } from 'react-intl';
-import ActivityFeed from './ActivityFeed/activity-feed/ActivityFeed';
 import SidebarContent from './SidebarContent';
 import messages from '../messages';
 import { withAPIContext } from '../APIContext';
 import { withErrorBoundary } from '../ErrorBoundary';
 import { getBadUserError, getBadItemError } from '../../util/error';
 import { DEFAULT_COLLAB_DEBOUNCE, ORIGIN_ACTIVITY_SIDEBAR } from '../../constants';
 import API from '../../api';
+import Logger from '../../logger';
 import './ActivitySidebar.scss';
+import Timer from '../../util/Timer'; // Note that this is a thin wrapper around performance api
+// To track whether or not the Activity Feed JS is loaded
+let isAFJSLoaded = false;
+const ActivityFeed = React.lazy(() => {
+    const afPromise = import('./ActivityFeed/activity-feed/ActivityFeed');
+    afPromise.then(() => (isAFJSLoaded = true));
```

Comment:
As far as I can tell, running something inside of the `then` is the only way we can know that the import has loaded. Note that Activity Feed will never load until first render occurs

Reply:
assuming the component will be created right away, could we move to the constructor?

Reply:
I'm doing away with this code entirely, and marking the "start" as ActivitySidebar constructor, and the Activity Feed marks are occurring inside the ActivityFeed itself (meaning that the AF JS has been loaded)